src/pages/en/guides/troubleshooting.md

```diff
@@ -0,0 +1,181 @@
+
+💡 Not sure that this is your problem? Check to see if anyone else has reported [this issue](https://github.com/withastro/astro/issues?q=is%3Aissue+is%3Aopen+Unable+to+render+Component)!
+
+
+### Expected a default export
+
+This error can be thrown when trying to import or render an invalid component, or one that is not working properly. (This particular message occurs because of the way importing a UI component works in Astro.)
+
+**Solution**: Try looking for errors in any component you are importing and rendering, and make sure it's working correctly. Consider opening an Astro starter template from [astro.new](https://astro.new) and troubleshooting just your component in a minimal Astro project.
+
+**Status**: Expected Astro behavior, as intended.
+
+## Common gotchas
+
+### My component is not rendering
+
+First, check to see that you have **imported the component** in your [`.astro` component script](/en/core-concepts/astro-components/#the-component-script) or [`.md` frontmatter](/en/guides/markdown-content/#using-components-in-markdown).
+
+Then check your import statement:
+
+- Is your import linking to the wrong place? (Check your import path.)
+
+- Does your import have the same name as the imported component? (Check your component name and that it [follows the `.astro` syntax](/en/comparing-astro-vs-other-tools/#astro-vs-jsx).)
+
+- Have you included the extension in the import? (Check that your imported file contains an extension. e.g. `.astro`, `.md`, `.jsx`, `.vue`)
+
+### My component is not interactive
+
+If your component is rendering (see above) but is not responding to user interaction, then you may be missing a [`client:*` directive](/en/reference/directives-reference/#client-directives) to hydrate your component.
+
+By default, a [UI Framework component is not hydrated in the client](/en/core-concepts/framework-components/#hydrating-interactive-components). If no `client:*` directive is provided, its HTML is rendered onto the page without JavaScript.
+
+Note: [Astro components](/en/core-concepts/astro-components/) are HTML-only templating components with no client-side runtime. But, you can use a `<script>` tag in your Astro component template to send JavaScript to the browser that executes in the global scope.
+
+### Cannot find package 'X'
+
+See [Astro's integration guide](/en/guides/integrations-guide/) for instructions on adding framework renderers, CSS tools and other packages to Astro.
+
+You may need to install peer dependencies for some integrations. If you see a “missing peer dependencies” warning, you can follow the instructions for  [handling dependencies](/en/guides/integrations-guide/#handling-integration-dependencies).
+
+### `Astro.glob()`'s "No matches found"
+
+When using `Astro.glob()` to import files, be sure to use the correct glob syntax that will match all the files you need.
+
+For example, use `src/components/**/*.js` to import both of the following files:
```

Comment:
The glob pattern _must_ be a relative path right? So actually `src/components/**/*.js` would not work. You need `../components/**/*.js` or something I think? Which makes the example trickier to show. Maybe:

```suggestion
For example, use `../components/**/*.js` in `src/pages/index.astro` to import both of the following files:
```

Reply:
So my question is, is this supposed to be demonstrating the proper relative path that goes INSIDE THE Astro.glob()

`Astro.glob('../components/**/*.js')` 

And if so, does the line make sense as: 

For example, use `Astro.glob('../components/**/*.js')` to import both of the following files:

Reply:
Because the path is relative, this pattern will only work in specific files.

The line does make sense, but if the file that contained it were nested differently alongside the `components` directory, it wouldn’t work.

- `Astro.glob('../components/**/*.js')` in `src/pages/index.astro` works
- `Astro.glob('../components/**/*.js')` in `src/pages/about/index.astro` doesn’t

Reply:
Got it. OK, then I'll go with your initial suggestion for simplicity! It gets the point across, and isn't meant to be comprehensive!